### PR TITLE
[SPARK-23958][CORE] HadoopRdd filters empty files to avoid generating empty tasks that affect the performance of the Spark computing performance.

### PR DESCRIPTION
**Pull request by guoxiaolongzte**
## What changes were proposed in this pull request?

HadoopRdd filter empty files to avoid generating empty tasks that affect the performance of the Spark computing performance.

Empty file's length is zero.

## How was this patch tested?

manual tests

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala [hunk 1]
```diff
@@ -55,7 +56,8 @@ private[spark] class HadoopPartition(rddId: Int, override val index: Int, s: Inp
 
   /**
    * Get any environment variables that should be added to the users environment when running pipes
-   * @return a Map with the environment variables and corresponding values, it could be empty
+    *
+    * @return a Map with the environment variables and corresponding values, it could be empty
```
Comment (Ngone51) [inline core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala]: nit: comment style

File: core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala [hunk 2]
```diff
@@ -86,8 +88,7 @@ private[spark] class HadoopPartition(rddId: Int, override val index: Int, s: Inp
  * @param keyClass Class of the key associated with the inputFormatClass.
  * @param valueClass Class of the value associated with the inputFormatClass.
  * @param minPartitions Minimum number of HadoopRDD partitions (Hadoop Splits) to generate.
- *
- * @note Instantiating this class directly is not recommended, please use
+  * @note Instantiating this class directly is not recommended, please use
```
Comment (Ngone51) [inline core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala]: ditto.

File: core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala (hunk 1 above)
Comment (guoxiaolongzte) [reply core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala]: what mean?

**Review by felixcheung — COMMENTED** [review]
yes, let's not change the comment.

could you add test for this?

**Review by HyukjinKwon — COMMENTED** [review]
Don't we already have a configuration for it?

https://github.com/apache/spark/blob/5fa438471110afbf4e2174df449ac79e292501f8/core/src/main/scala/org/apache/spark/internal/config/package.scala#L321-L326

**Comment by jerryshao** [comment]
Yes, this is already supported in Spark, seems like the PR is invalid.

**Comment by jiangxb1987** [comment]
@guoxiaolongzte Have you tried the config `spark.hadoopRDD.ignoreEmptySplits` ?

**Comment by guoxiaolongzte** [comment]
1.No need to loop twice to filter to determine if the length is greater than 0 
2.This feature is to improve performance, the default switch needs to open

**Pull request closed without merging**